File: java/app/src/main/java/net/fbvictorhugo/j/barreirasanitaria/data/dao/PessoaDAO.java
package net.fbvictorhugo.j.barreirasanitaria.data.dao;

import net.fbvictorhugo.j.barreirasanitaria.data.model.Pessoa;

import java.util.List;

public interface PessoaDAO {

    List<Pessoa> listar();

    void inserir(Pessoa pessoa) throws Exception;

    void atualizar(Pessoa pessoa) throws Exception;

    List<Pessoa> pesquisarPorDocumento(long numeroDocumento);

    Pessoa procurarPessoa(long pessoaId);

}

File: kotlin/app/src/main/java/net/fbvictorhugo/k/barreirasanitaria/data/dao/PessoaDAO.kt
package net.fbvictorhugo.k.barreirasanitaria.data.dao

import net.fbvictorhugo.k.barreirasanitaria.data.model.Pessoa

interface PessoaDAO {

    fun listar(): ArrayList<Pessoa>

    @Throws(Exception::class)
    fun inserir(pessoa: Pessoa)

    @Throws(Exception::class)
    fun atualizar(pessoa: Pessoa)

    fun pesquisarPorDocumento(numeroDocumento: Long): ArrayList<Pessoa>

    fun procurarPessoa(pessoaId: Long): Pessoa?

}
